backend/laundry/api_wrapper.py

```diff
@@ -1,46 +1,58 @@
 import requests
-from bs4 import BeautifulSoup
 from django.conf import settings
 from django.utils import timezone
 from requests.exceptions import ConnectTimeout, HTTPError, ReadTimeout
 
 from laundry.models import LaundryRoom, LaundrySnapshot
 
 
-HALL_URL = f"{settings.LAUNDRY_URL}/?location="
+def get_room_url(room_id: int):
+    return f"{settings.LAUNDRY_URL}/rooms/{room_id}/machines?raw=true"
 
 
-def update_machine_object(cols, machine_object):
+headers = {
+    "x-api-key": settings.LAUNDRY_X_API_KEY,
+    "alliancels-auth-token": settings.LAUNDRY_ALLIANCELS_API_KEY,
+}
+
+
+def update_machine_object(machine, machine_type_data):
     """
     Updates Machine status and time remaining
     """
 
-    if cols[2].getText() in ["In use", "Almost done"]:
-        time_remaining = cols[3].getText().split(" ")[0]
-        machine_object["running"] += 1
+    #  TODO: Early stage in update 9/29/2024, known status codes are "IN_USE", "AVAILABLE", "COMPLETE"; need to update
+    #  TODO: if we identify other codes, especially error
+    status = machine["currentStatus"]["statusId"]
+    if status == "IN_USE":
+        time_remaining = machine[3].getText().split(" ")[0]
+        machine_type_data["running"] += 1
         try:
-            machine_object["time_remaining"].append(int(time_remaining))
+            machine_type_data["time_remaining"].append(int(time_remaining))
         except ValueError:
             pass
-    elif cols[2].getText() == "Out of order":
-        machine_object["out_of_order"] += 1
-    elif cols[2].getText() == "Not online":
-        machine_object["offline"] += 1
+    elif status in ["AVAILABLE", "COMPLETE"]:
+        machine_type_data["open"] += 1
+    # elif status == "Out of order":
```

Comment:
let's remove this dead code

Reply:
I left this here so we know what some of the legacy code does. Want to full rewrite at some point, but this is in case we find out these statuses work with the new machines too.